# **Experiment Tracking with MLflow**

## Q1. Install MLflow

In [1]:
!mlflow --version

mlflow, version 1.26.1


## Q2. Download and preprocess the data

In [2]:
!python preprocess_data.py --raw_data_path data/ --dest_path output/

In [3]:
ls output

 Volume in drive E is Data
 Volume Serial Number is 5630-3F58

 Directory of e:\Justin\Documents\Codes\data-talks-club\mlops-zoomcamp\02-experiment-tracking\output

30/05/2022  08:35 pm    <DIR>          .
30/05/2022  08:35 pm    <DIR>          ..
30/05/2022  08:39 pm           305,256 dv.pkl
30/05/2022  08:39 pm         2,805,197 test.pkl
30/05/2022  08:39 pm         2,661,149 train.pkl
30/05/2022  08:39 pm         2,166,527 valid.pkl
               4 File(s)      7,938,129 bytes
               2 Dir(s)  32,199,675,904 bytes free


How many files were saved to `OUTPUT_FOLDER`?
> **Answer: 4**

## Q3. Train a model with autolog

In [4]:
!python train.py --data_path output/

train.py


2022/05/30 20:39:36 INFO mlflow.tracking.fluent: Autologging successfully enabled for sklearn.


How many parameters are automatically logged by MLflow?
> **Answer: 17**

## Q5. Tune the hyperparameters of the model


In [5]:
!python hpo.py --data_path output/

hpo.py

100%|██████████| 50/50 [20:19<00:00, 24.40s/trial, best loss: 6.6284257482044735]


2022/05/30 20:59:02 INFO mlflow.tracking.fluent: Experiment with name 'random-forest-hyperopt' does not exist. Creating a new experiment.


What's the best validation RMSE that you got?
> **Answer: 6.628**

## Q6. Promote the best model to the model registry

In [6]:
!python register_model.py --data_path output/

register_model.py
Amount of runs: 5
Best run ID: runs:/ae9b2920602c451c8183c4a91ce1f6c0/models


2022/05/30 21:25:14 INFO mlflow.tracking.fluent: Experiment with name 'random-forest-best-models' does not exist. Creating a new experiment.
Successfully registered model 'RandomForestRegressionModel'.
2022/05/30 21:28:29 INFO mlflow.tracking._model_registry.client: Waiting up to 300 seconds for model version to finish creation.                     Model name: RandomForestRegressionModel, version 1
Created version '1' of model 'RandomForestRegressionModel'.


What is the test RMSE of the best model?
> **Answer: 6.55**